core/lib/spree/event/adapters/active_support_notifications.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Spree
+  module Event
+    module Adapters
+      module ActiveSupportNotifications
```

Comment:
This is more of a question and less of a request. Would it be worth implementing more of an interface or strategy pattern? So instead of defining a module, have a base abstract class which all adapters should inherit from an therefore expected to implement `instrument`, `subscribe`, etc.

I know I'm splitting hairs here but I love to learn

Reply:
I guess it mostly depends on specific situations. The main advantage of using an adapter is that you can easily change the adapter at runtime and still call the same interface and get the work done. 

If you are thinking about any specific implementation that may work better in this situation then please share it, all improvements are welcome :)

Reply:
I think this is great, I say splitting hairs because they all accomplish the same exact thing. Some are just more strictly enforced than others